### ISSUE TITLE
Highcharts Editor bar chart empty when embedding on website.

### ISSUE THREAD

**Issue by vanherzeelel** [highcharts/highcharts#21118]
#### Expected behaviour
The bar chart should not be empty when embedding the HTML output.

#### Actual behaviour
In the browser (Highcharts Editor), we see the full graph (including the bars), but on the website, we see a formatter related browser error (empty chart).

Browser:
![Snag_23377bce](https://github.com/highcharts/highcharts/assets/134702063/a0cdc541-4970-40e6-b6c8-25d46feb2058)

Website:
![Snag_2337adbb](https://github.com/highcharts/highcharts/assets/134702063/a89f8036-b652-457a-a921-f4642e18aa47)

#### Live demo with steps to reproduce
Please find the HTML output of the chart in attachment.
[EMEExChina.txt](https://github.com/highcharts/highcharts/files/15218828/EMEExChina.txt)

#### Product version
[](https://editor.highcharts.com/full.html)

#### Affected browser(s)
Chrome

**Comment by karolkolodziej** [comment]
Hi @vanherzeelel! 

I was able to recreate it by only enabling the data labels for the series- [see in the demo.](https://jsfiddle.net/BlackLabel/8nyqsevw/)

Issue here is that the `formatter` can't be serialised and for some reason by default the `formatter` in the HTML is set as:
```js
"formatter": "function(){let{numberFormatter:t}=this.series.chart;return\"number\"!=typeof this.y?\"\":t(this.y,-1)}",
```

Whereas if you use JSON the `formatter` isn't set here- [see demo.](https://jsfiddle.net/BlackLabel/hk03m51r/)

As a workaround you can either:
- [remove/comment the formatter from the HTML generated code](https://jsfiddle.net/BlackLabel/khnp8aj2/)
- [use generated JSON and build the chart on your own](https://jsfiddle.net/BlackLabel/hk03m51r/)

**Comment by DaanVanDeVelde** [comment]
Hello @karolkolodziej, 
Commenting/removing the formatter does seem to resolve the issue. However it's not that user friendly to always have to do it manually, especially if we have charts like these with 10+ datalabels, we have to do a find and replace in the html for each of them. Is there a setting we can use in the editor to do this more easily?

**Comment by karolkolodziej** [comment]
@DaanVanDeVelde Have you tried exporting the chart to JSON?
![image](https://github.com/highcharts/highcharts/assets/57004120/e07b77de-42e1-4a61-9ba4-8093521e1502)

You can follow [the documentation](https://www.highcharts.com/docs/getting-started/your-first-chart) on creating a chart if you have your options.
Please refer to the demo I've created with JSON.

**Comment by DaanVanDeVelde** [comment]
Hello @karolkolodziej, Our current implementation doesn't allow for the usage of JSON. I conclude that this is an actual bug with highcharts. Do we leave this bug open so it can be fixed in the future?

**Comment by karolkolodziej** [comment]
You are right the HTML config should not contain the `formatter` if not set. Let's keep this open.
